include/rocksdb/utilities/object_registry.h

```diff
@@ -108,9 +206,18 @@ class ObjectLibrary {
   template <typename T>
   const FactoryFunc<T>& Register(const std::string& pattern,
                                  const FactoryFunc<T>& factory) {
-    std::unique_ptr<Entry> entry(new FactoryEntry<T>(pattern, factory));
-    AddEntry(T::Type(), entry);
-    return factory;
+    PatternEntry entry(pattern);
+    return Register(entry, factory);
+  }
+
+  template <typename T>
+  const FactoryFunc<T>& Register(const PatternEntry& pattern,
```

Comment:
`Entry` appears to be publicly extensible, inviting custom matching implementations, which I don't really object to except that we are also using value semantics (copy) for a kind of Entry here. At least in public facing API, you should be consistent about value vs. owned pointer semantics for the entire hierarchy. IMHO mixing extensibility with value semantics in a hierarchy is confusing (because you obviously can't have both in the same place at the same time).